src/runtime/complex.cpp

```diff
@@ -517,398 +417,172 @@ Box* complexStr(BoxedComplex* self) {
     if (!PyComplex_Check(self))
         raiseExcHelper(TypeError, "descriptor '__str__' requires a 'complex' object but received a '%s'",
                        getTypeName(self));
-    Box* r = complex_fmt(self->real, self->imag, 12, 'g');
+
+    // make sure that BoxedComplex::imag/real have the same offsets as
+    // CAPI PyComplexObject::cval.imag/real
+    static_assert(offsetof(BoxedComplex, real) == offsetof(PyComplexObject, cval.real), "");
```

Comment:
Can we please move this into ```runtime/types.h``` directly after the ```BoxedComplex``` declaration like we already do for other types (eg. BoxedFloat) and add a additional ```static_assert(sizeof(BoxedComplex) == sizeof(PyComplexObject), "")``` assert.
sorry that I did not make this clear from the beginning :-(.